org.opent2t.sample.multisensor.superpopular/com.wink.sensorpod/js/thingTranslator.js

```diff
@@ -70,10 +70,15 @@ class StateReader {
 }
 
 function convertDeviceDateToTranslatorDate(unixTimestamp) {
-    // Date takes a number of milliseconds, so convert seconds to milliseconds
-    var datetime = new Date(unixTimestamp * 1000);
+    // Only convert if input is valid numeric value
+    if (unixTimestamp && !isNaN(unixTimestamp)) { 
+        // Date takes a number of milliseconds, so convert seconds to milliseconds
+        var datetime = new Date(unixTimestamp * 1000);
+        return datetime.toISOString();
+    }
 
-    return datetime.toISOString();
+    // return all other invalid input as-is since we can't convert to DateTime
+    return unixTimestamp;
```

Comment:
WINK returns "undefined" - a string value - in some cases only not always - while its great to understand why WINK doesnt adhere to the standard i think our own code should also be robust here and handle at least the values that we are observing are being sent back and not blowing up. Here the issue is the Date constructor has been working for null returning an epoch value date; so us returning what the 3rd party provider is returning (whether they are buggy ornot ) seems to be least controversial and cooked up to me at hte moment. 

---
In reply to: [106047060](https://github.com/openT2T/translators/pull/210#discussion_r106047060) [](ancestors = 106047060)